core/src/main/java/org/elasticsearch/index/query/HasChildQueryParser.java

```diff
@@ -99,10 +98,8 @@ public Query parse(QueryParseContext parseContext) throws IOException, QueryPars
             } else if (token.isValue()) {
                 if ("type".equals(currentFieldName) || "child_type".equals(currentFieldName) || "childType".equals(currentFieldName)) {
                     childType = parser.text();
-                } else if ("score_type".equals(currentFieldName) || "scoreType".equals(currentFieldName)) {
-                    scoreType = ScoreType.fromString(parser.text());
```

Comment:
can you push a commit to 2.x that marks score_type as deprecated? ideally using ParseField